packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1147,14 +1147,23 @@ class TextSelectionGestureDetectorBuilder {
             case PointerDeviceKind.unknown:
               // On macOS/iOS/iPadOS a touch tap places the cursor at the edge
               // of the word.
-              renderEditable.selectWordEdge(cause: SelectionChangedCause.tap);
+              final TextSelection? previousSelection = renderEditable.selection;
+              final TextSelection newSelection = renderEditable.selectWordEdge(cause: SelectionChangedCause.tap);
```

Comment:
Rather than changing selectWordEdge to return the new selection, could you check renderEditable.selection afterwards?

Reply:
Yep, that makes much more sense. Updated.